### PR TITLE
New: Elide adapter for sending fact requests

### LINKED ISSUE #854: Implement navi async elide facts adapter

### PR DESCRIPTION
**Pull request by mattgraham1995**
Resolves #854 

## Proposed Changes

- Create methods that will kick off upsert and cancel queries for AsyncRequests in Elide

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/adapters/elide-facts.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      variables: null
+    });
+  }
+
+  /**
+   * @param request
+   * @param options
+   * @returns Promise that resolves to the result of the AsyncQuery creation mutation
+   */
+  asyncFetchDataForRequest(request: RequestV1, options: RequestOptions): Promise<AsyncQueryResponse> {
```
Comment (jkusa) [inline packages/data/addon/adapters/elide-facts.ts:65]:
```suggestion
  createAsyncQueryRequest(request: RequestV1, options: RequestOptions): Promise<AsyncQueryResponse> {
```

File: packages/data/addon/adapters/fact-interface.ts [hunk 2]
```diff
@@ -71,18 +71,39 @@ export enum QueryStatus {
...
+  httpStatus: number;
+  contentLength: number;
+  responseBody: string;
+}
+
 export default interface NaviFactAdapter {
   fetchDataForRequest(request: RequestV1 | RequestV2, options: RequestOptions): Promise<TODO>;
   urlForFindQuery(request: RequestV1 | RequestV2, options: RequestOptions): string;
-  asyncFetchDataForRequest(request: RequestV1 | RequestV2, options: RequestOptions): AsyncQuery;
+  asyncFetchDataForRequest(request: RequestV1 | RequestV2, options: RequestOptions): Promise<AsyncQueryResponse>;
```
Comment (jkusa) [inline packages/data/addon/adapters/fact-interface.ts:108]:
```suggestion
```
Let's come back once the contract is settled.

File: packages/data/addon/adapters/elide-facts.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+interface ApolloMutationOptions {
+  variables?: Record<string, any>;
+  mutation: DocumentNode;
+}
+interface RequestMetric {
+  metric: string;
+  parameters?: Dict<string>;
+}
+interface RequestDimension {
```
[the remark this answers is not in the document]
Comment (daniellevinson) [reply packages/data/addon/adapters/elide-facts.ts:20]: We do but they're in the request-v2 branch, not in master

File: packages/data/addon/adapters/elide-facts.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    const mutation: DocumentNode = GQLQueries['asyncFactsCancel'];
+    return this.apollo.mutate({
+      mutation,
+      variables: {
+        ids: [id]
+      }
+    });
+  }
+
+  urlForFindQuery(_request: RequestV1, _options: RequestOptions): string {
```
Comment (daniellevinson) [inline packages/data/addon/adapters/elide-facts.ts:96]: jsdoc for these

**Pull request merged**